controlplane/pkg/nsm/nsm_heal_processor.go

```diff
@@ -108,6 +118,22 @@ func (p *healProcessor) Heal(ctx context.Context, clientConnection nsm.ClientCon
 
 func (p *healProcessor) CloseConnection(ctx context.Context, conn nsm.ClientConnection) error {
 	var err error
+
+	// Cancell context after closing.
+	defer func() {
+		var contextCancelFunc func()
+		var isHealing bool
+
+		p.healCancellersMutex.Lock()
+		contextCancelFunc, isHealing = p.healCancellers[conn.GetID()]
```

Comment:
`contextCancelFunc, isHealing := p.healCancellers[conn.GetID()]` instead of `var` declarations